_drafts/SC3-3-2-form-field-has-name.md

```diff
@@ -0,0 +1,77 @@
+---
+name: Form field has accessible name
+description: |
+  Each form field element has an accessible name
+
+success_criterion:
+- 3.3.2 # Labels or Instructions
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Anne Thyme Nørregaard
+- Bryn Anderson
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to:
+- any element that is [exposed to assistive technologies](#exposed-to-assistive-technologies), and
+- that has a [semantic role](#semantic-role) that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
+- does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+**Note**: All elements that are semantic form controls, either implicitly through native HTML elements or explicitly through use of ARIA roles, inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` roles. The required context role defines the owning container in which a given role is allowed. Form controls with a required context role that is itself a form control, e.g. `<option>` within a `<select>`, are not applicable to this rule as only the top-most form control requires an accessible name.
+
+**Note**: At the time of writing this rule the semantic roles that inherit from the abstract `input` or `select` role and do not have a required context role that itself inherits from one of those roles are: `checkbox`, `combobox`, `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider, spinbutton`, `switch`, `textbox`.
```

Comment:
There is no "at the time of writing" - this is what the ARIA 1.1 spec says, that won't change. Maybe ARIA 1.2 comes along and this changes, or some module is added. If I write my own ARIA module tomorrow, inherits from the ARIA 1.1 select role, that doesn't immediately mean it gets included in this list. This needs to be unambiguous - we can't be arguing over who's spec we consider inheritance from.